### PR TITLE
refactor: drop lodash.merge

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

#3753

This is only a PoC of lodash.merge replacement. If this PR doesn't have huge performance impact, we should bring up a `deepMerge()` in `hexo-util`.

## How to test

```sh
git clone -b drop-lodash-for-config https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

**Comment by SukkaW** [comment]
With https://github.com/hexojs/hexo-util/pull/141 we could replace all `lodash.merge` with `hexo-util.deepMerge`.

**Comment by curbengh** [comment]
https://github.com/hexojs/hexo-util/pull/141 has been merged, not released yet. Can this PR uses master branch of hexo-util temporarily for testing? Once that PR is released, simply update the hexo-util dep.

File: package.json [hunk 1]
```diff
@@ -44,12 +44,13 @@
...
     "chalk": "^3.0.0",
     "cheerio": "0.22.0",
+    "deepmerge": "^4.2.2",
     "hexo-cli": "^3.0.0",
     "hexo-front-matter": "^1.0.0",
     "hexo-fs": "^2.0.0",
     "hexo-i18n": "^1.0.0",
     "hexo-log": "^1.0.0",
-    "hexo-util": "^1.5.0",
+    "hexo-util": "github:hexojs/hexo-util#master",
```
Comment (curbengh) [inline package.json]: 1.7.0 released

File: package.json [hunk 2]
```diff
@@ -44,12 +44,13 @@
     "bluebird": "^3.5.2",
     "chalk": "^3.0.0",
     "cheerio": "0.22.0",
+    "deepmerge": "^4.2.2",
```
Comment (curbengh) [inline package.json]: necessary? or it's for unit test?

File: package.json (hunk 1 above)
Comment (SukkaW) [reply package.json]: Rebased.

File: package.json (hunk 2 above)
Comment (SukkaW) [reply package.json]: Rebased and it is now removed.

**Comment by curbengh** [comment]
The soft fail is addressed in https://github.com/hexojs/hexo/pull/3924

**Pull request merged**